docs/source/configuration/authorization.mdx

```diff
@@ -0,0 +1,302 @@
+
+Depending on which scope set is presented by the request, different parts of the query may be available.
+
+Assuming we have this schema:
+
+```graphql
+type Query {
+    me: User @requiresScopes(scopes: ["profile:read"])
+    products: [Products]
+}
+
+type User {
+    id: ID
+    name: String
+    email: @requiresScopes(["user:read"])
+}
+
+type Product {
+    id: ID
+    name: String
+    amount: Int @requiresScopes(["employee", "inventory"])
+    orders: [User] @requiresScopes(["employee"])
+}
+```
+
+And this query:
+
+```graphql
+{
+    me {
+        name
+        email
+    }
+
+    products {
+        name
+        amount
+        orders {
+            name
+        }
+    }
+}
+```
+
+If the request presented the scopes `profile:read user:read`, then it would be interpreted as:
+
+
+```graphql
+{
+    me {
+        name
+        email
+    }
+
+    products {
+        name
+    }
+}
+```
+
+And generate an error at path `/products/@/amount`.
+
+While if it presented the scopes `employee inventory`, it would filter the query as:
+
+```graphql
+{
+    products {
+        name
+        amount
+        orders {
+            name
+        }
+    }
+}
+```
+
+And generate an error at path `/me`.
+
+
+## Composition and federation
+
+The authorization directives are defined by the subgraph author, and will be carried into the supergraph schema:
+- `@authenticated`: if defined on a field or type by any of the subgraphs, it will be set in the supergraph too
+- `@requiresScopes`: same as `@authenticated`, and if there are multiple applications, the supergraph schema will merge the sets of scopes required by each subgraph
+
+### Authorization and `@key`
+
+When the authorization directives are set on fields used in `@key`, they will be usable by the router to join between subgraphs, but cannot be queried directly by the client. This behavior is similar to [Contracts](/graphos/delivery/contracts/).
+
+As an example, assuming we have these subgraphs definition:
+
+```graphql
+# subgraph A
+type Query {
+  t: T
+}
+
+type T @key(fields: "id") {
+  id: ID! @authenticated
+}
+```
+
+```graphql
+# subgraph B
+type T @key(fields: "id") {
+  id: ID! @authenticated
+  value: Int
+}
+```
+
+The following query can be done when not authenticated:
+
+```graphql
+{
+  t {
+    value
+  }
+}
+```
+
+But this one would end up with a `null` in `id`, which would trigger nullability rules and nullify `t` too.
+
+```graphql
+{
+  t {
+    id
+    value
+  }
+}
+```
+
+### Interfaces
+
+In the case one of the types implementing an interface requires authorization, then querying the interface will be allowed, but any parts that require access to that type will be filtered.
+
+As an example, with this schema, where the interface  `I` does not require authentication, but the `User` type does:
+
+```graphql
+type Query {
+  itf: I!
+}
+
+interface I {
+    id: ID
+}
+
+type User
+implements I
+@authenticated {
```

Comment:
this is the only example that applies the directive to a type. should we have a section on "inheritance rules" and how applying these directives to types affect fields across subgraphs? (this is a common footgun in contracts)

Reply:
I think so yes. I'm unsure about the balance in these docs between simple examples and nearly full spec describing edge cases. I think we'll end up documenting all the edge cases here anyway